### PR TITLE
[RFC] vim-patch:7.4.925

### PR DESCRIPTION
**Pull request by watiko**
```
Problem:    User may yank or put using the register being recorded in.
Solution:   Add the recording register in the message. (Christian Brabandt,
            closes #470)
```

https://github.com/vim/vim/commit/a0ed84a26897c994512873a895b9fc54e90c6845

### CODE REVIEW

**Comment by ZyX-I** [comment]
Should not we do something with “closes #470”? *Our* #470 has nothing to do with what is referenced in the commit message.

I think that vim-patch.sh should at least change this to `vim/vim#470`.

**Comment by justinmk** [comment]
> I think that vim-patch.sh should at least change this to vim/vim#470

:+1:

File: src/nvim/option_defs.h [hunk 1]
```diff
@@ -151,26 +151,39 @@
...
-#define SHM_WRITE       'W'             /* don't use "written" at all */
-#define SHM_TRUNC       't'             /* trunctate file messages */
-#define SHM_TRUNCALL    'T'             /* trunctate all messages */
-#define SHM_OVER        'o'             /* overwrite file messages */
-#define SHM_OVERALL     'O'             /* overwrite more messages */
-#define SHM_SEARCH      's'             /* no search hit bottom messages */
-#define SHM_ATTENTION   'A'             /* no ATTENTION messages */
-#define SHM_INTRO       'I'             /* intro messages */
-#define SHM_COMPLETIONMENU 'c'          // completion menu messages
-#define SHM_ALL         "rmfixlnwaWtToOsAIc" /* all possible flags for 'shm' */
```
Comment (watiko) [inline src/nvim/option_defs.h]: `SHM_ALL` contains `'a'` so we need add `'a'` somehow.

Comment (ZyX-I) [inline src/nvim/option_defs.h]: `option.c`, function `shortmess` contains `vim_strchr(p_shm, 'a')`, it is also the only other place where SHM_A is used. I would suggest:

1. Rename SHM_A to SHM_ALL_ABBREVIATIONS (SHM_ALL + _ABBREVIATIONS).
2. Add `SHM_ABBREVIATIONS = 'a',  ///< Use abbreviations from #SHM_ALL_ABBREVIATIONS` to the enum.
3. Use this constant in place of `'a'` in `shortmess` function.

Comment (watiko) [inline src/nvim/option_defs.h]: Thanks a lot!

**Pull request merged**